via/services/youtube.py

```diff
@@ -39,6 +41,15 @@ def get_video_id(self, url):
 
         return None
 
+    def get_transcript(self, video_id):
+        """
+        Call the YouTube API and return the transcript for the given video_id.
+
+        :raise Exception: this method might raise any type of exception that
+            YouTubeTranscriptApi raises
+        """
+        return YouTubeTranscriptApi.get_transcript(video_id)
```

Comment:
I think we might be shooting ourselves in the caching foot by putting this in the same service as the other stuff.

I think the ideal here is to have a well managed global cache (not a per request cache which will be pretty much useless).

We could do that by instantiating this whole service as a singleton. The only non-singleton thing it has is the `HTTPService` which is pretty much a fancy wrapper around `requests.Session`. You can instantiate this directly, or have it be a singleton too, but I seem to recall we've had this discussion and the general feeling was of possible heeby jeebies around long lived / shared sessions. I think that the session does it's own connection pooling, management and recycling etc, but I can't prove the negative, so we can leave that discussion as is.

The other way to do it is to have a separate singleton service around just the transcript. So we'd have a `YouTubeTranscriptService` (singleton) and a `YouTubeInfoService` (per request). This way you could add a nice TTL cache or whatever we'd like to reduce the amount of noise we send to YouTube.

Reply:
Yes, we're going to adding Postgres, that'll be our global cache